kuwala/core/canvas/src/components/Explorer/Selector/SelectorSchemaExplorer.js

```diff
@@ -0,0 +1,281 @@
+                            />
+                        </span>
+                            <span className={'font-semibold text-md'}>
+                            {el.category}
+                        </span>
+                        </div>
+                        {el.isOpen ?
+                            el.tables.map(el => generateCategoryTables(el, currentKey))
+                            : null
+                        }
+                    </div>
+                )
+            })
+        )
+    }
+
+    const generateCategoryTables = (tableName, parent) => {
+        const tableKey = `${parent}@${tableName}`
+        return (
+            <div
+                className={`
+                    flex flex-row items-center pl-20 pr-8 py-2
+                    cursor-pointer
+                    min-w-max
+                    ${tableKey === selectedTable ? `bg-kuwala-green text-white` : `bg-white text-black`}
+                `}
+                key={tableKey}
+                onClick={()=>{
+                    tableSelectorOnclick({
+                        addressString: tableKey,
+                    })
+                }}
+            >
+                <span className={'mr-4'}>
+                    <img
+                        src={TableSVG}
+                        style={{minWidth: 16, minHeight: 16}}
+                    />
+                </span>
+                <span className={'font-semibold text-md w-full'}>
+                    {tableName}
+                </span>
+            </div>
+        )
+    }
+
+    const toggleTreeItem = (addressString) => {
+        const arr = addressString.split('@')
+        const schemaAddress = arr[0]
+        const categoryAddress = arr[1]
+
+        let tempSchema;
+        if(categoryAddress && schemaAddress) {
+            tempSchema = schemaList.map((el) => {
+                if (el.schema === schemaAddress) {
+                    return {
+                        ...el,
+                        categories: el.categories.map((cat) => {
+                            if (cat.category === categoryAddress){
+                                cat.isOpen = !cat.isOpen
+                            }
+                            return cat
+                        })
+                    }
+                }
+                return el
+            })
+        } else {
+            tempSchema = schemaList.map((el) => {
+                if (el.schema === schemaAddress){
+                    el.isOpen = !el.isOpen
+                    if (el.isOpen === false) {
+                        return {
+                            ...el,
+                            categories: el.categories.map((cat) => {
+                                cat.isOpen = false
+                                return cat
+                            })
+                        }
+                    }
+                }
+                return el
+            })
+        }
+        setSchema(tempSchema)
+    }
```

Comment:
Feels to me like this is all DRY code. Please refactor with the data source preview components.